File: models/product_model.js
import { Schema, model } from 'mongoose';
//import { v1 } from 'uuid';
//import { genSaltSync, hashSync } from 'bcrypt';

const productSchema = Schema({
    category: { type: Schema.Types.ObjectId, ref: 'Category', required: true, },
    title: { type: String, required: [true, "title is required, pls provide"] },
    description: { type: String, default: '', },
    price: { type: Number, required: true, },
    images: { type: Array, default: [], },
    updatedOn: { type: Date, },
    createdOn: { type: Date, }

});

productSchema.pre('save', function (next) {
    this.updatedOn = new Date();
    this.createdOn = new Date();

    next(); // if next has string that mean it has error
});

productSchema.pre(['update', 'findOneAndUpdate'], function (next) {
    const update = this.getUpdate();

    this.updatedOn = new Date();
    next();
});

const productModel = model('Product', productSchema);

//module.exports = productModel;
export default productModel;

File: ecommerce_frontend/lib/data/models/product_model.dart
class ProductModel {
  String? sId;
  late String category;
  late String title;
  late String description;
  late num price;
  late List<String> images;
  String? updatedOn;
  String? createdOn;

  ProductModel(
      {this.sId,
      required this.category,
      required this.title,
      required this.description,
      required this.price,
      required this.images,
      this.updatedOn,
      this.createdOn});

  ProductModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    category = json['category'];
    title = json['title'];
    description = json['description'];
    price = json['price'];
    images = json['images'].cast<String>();
    updatedOn = json['updatedOn'];
    createdOn = json['createdOn'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = sId;
    data['category'] = category;
    data['title'] = title;
    data['description'] = description;
    data['price'] = price;
    data['images'] = images;
    data['updatedOn'] = updatedOn;
    data['createdOn'] = createdOn;
    return data;
  }
}
